generic/tests/fillup_disk.py

```diff
@@ -20,45 +23,34 @@ def run(test, params, env):
     vm.verify_alive()
     login_timeout = int(params.get("login_timeout", 360))
     session = vm.wait_for_login(timeout=login_timeout)
-    session2 = vm.wait_for_serial_login(timeout=login_timeout)
 
-    fillup_timeout = int(params.get("fillup_timeout"))
-    fillup_size = int(params.get("fillup_size"))
-    fill_dir = params.get("guest_testdir", "/tmp")
+    disk_size = params.get("image_size_disk1")
+    mountpoint = params.get("mountpoint")
+    disk_idx = params.get("disk_index", 1)
+    fstype = params.get("fstype", "ext3")
+
+    error.context("Format the disk in %s" % mountpoint, logging.info)
+    utils_misc.format_linux_disk(session, mountpoint,
+                                 disk_idx, disk_size, fstype)
+
     filled = False
     number = 0
-
+    fillup_timeout = int(params.get("fillup_timeout"))
+    cmd = params.get("fillup_cmd")
     try:
-        error.context("Start filling the disk in %s" % fill_dir, logging.info)
-        cmd = params.get("fillup_cmd")
+        error.context("Start filling the disk in %s" % mountpoint, logging.info)
         while not filled:
             # As we want to test the backing file, so bypass the cache
-            tmp_cmd = cmd % (fill_dir, number, fillup_size)
-            logging.debug(tmp_cmd)
-            s, o = session.cmd_status_output(tmp_cmd, timeout=fillup_timeout)
-            if "No space left on device" in o:
+            fillup_cmd = cmd % number
+            status, output = session.cmd_output(fillup_cmd,
```

Comment:
Yes, it did.

in debug log:
07/28 13:24:55 DEBUG|   aexpect:1465| Sending command: dd if=/dev/zero of=/media/fillup.4 bs=200M count=1 oflag=direct
07/28 13:24:58 DEBUG|   aexpect:1465| Sending command: echo $?
07/28 13:24:58 DEBUG|   aexpect:1465| Sending command: dd if=/dev/zero of=/media/fillup.5 bs=200M count=1 oflag=direct